sdk/servicebus/azure-servicebus/azure/servicebus/amqp/_amqp_message.py

```diff
@@ -233,14 +249,18 @@ def _to_outgoing_amqp_message(self):
                 correlation_id=self.properties.correlation_id,
                 content_type=self.properties.content_type,
                 content_encoding=self.properties.content_encoding,
-                creation_time=int(self.properties.creation_time) if self.properties.creation_time else None,
-                absolute_expiry_time=int(self.properties.absolute_expiry_time)
-                if self.properties.absolute_expiry_time else None,
+                creation_time=creation_time_from_ttl if ttl_set else None,
+                absolute_expiry_time=absolute_expiry_time_from_ttl if ttl_set else None,
```

Comment:
isn't that we also need to check whether `self.properties.creation_time` exists?

```python
if ttl_set:
    creation_time = creation_time_from_ttl 
elif self.properties.creation_time:
    creation_time = int(self.properties.creation_time)
else:
    creation_time = None
```

same applies to `absolute_expiry_time`